src/Definition/Reflection/ClassDefinition.php

```diff
@@ -69,7 +71,10 @@ public function getInterfaces(): array
         return $this->reflection->getInterfaceNames();
     }
 
-    private function reflectParameters()
+    /**
+     * @psalm-assert array<string, Parameter> $this->parameters
```

Comment:
instead of this, wouldn't it be better to simply set `parameters` to array with a default value?

Reply:
Probably yes, but this would require additional refactoring to this class since `null` is used by getter methods. I just added the types to reflect the existing behaviour.